lib/iris/tests/unit/lazy_data/test_multidim_daskarray.py

```diff
@@ -0,0 +1,58 @@
+# (C) British Crown Copyright 2017, Met Office
+#
+# This file is part of Iris.
+#
+# Iris is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Iris is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Iris.  If not, see <http://www.gnu.org/licenses/>.
+"""Test function :func:`iris._lazy data.multidim_daskstack`."""
+
+from __future__ import (absolute_import, division, print_function)
+from six.moves import (filter, input, map, range, zip)  # noqa
+
+# Import iris.tests first so that some things can be initialised before
+# importing anything else.
+import iris.tests as tests
+
+import numpy as np
+import dask.array as da
+
+from iris._lazy_data import multidim_daskstack
+
+
+class Test_multidim_daskstack(tests.IrisTest):
+    def test_0d(self):
+        value = 4
+        data = np.array(da.from_array(np.array(value), chunks=1), dtype=object)
+        result = multidim_daskstack(data)
+        self.assertEqual(result, value)
+
+    def test_1d(self):
+        vals = [4, 11]
+        data = np.array([da.from_array(val*np.ones((3, 3)), chunks=3) for val
+                         in vals], dtype=object)
+        result = multidim_daskstack(data)
+        self.assertEqual(result.shape, (2, 3, 3))
+        self.assertArrayEqual(result[:, 0, 0], np.array(vals))
+
+    def test_2d(self):
+        vals = [4, 8, 11]
+        data = np.array([da.from_array(val*np.ones((2, 2)), chunks=2) for val
+                         in vals], dtype=object)
+        data = data.reshape(3, 1, 2, 2)
```

Comment:
I'd argue this data is no longer 2D! Perhaps you could rename this test to `test_nd`?